scripts/run_benchmarks.py

```diff
@@ -209,11 +267,15 @@ def main(args: argparse.Namespace):
                 },
                 allow_val_change=True,
             )
-            wandb_run.log(results["results"] | lb_results)
+            wandb_run.log(results | lb_results)
             wandb_run.finish()
 
-            store.add(state)
-            store.save()
+            last_model = state
+            save_state(last_model, args.file)
+
+            if step % 50:
```

Comment:
Could probably check if the cache is over a certain size too although this is fine.

Reply:
Going with the simpler option here since we aren't likely to reuse a model in the cache anyway.